matrix-api.c

```diff
@@ -210,6 +217,81 @@ static int _handle_headers_complete(http_parser *http_parser)
 }
 
 
+static gchar *_gunzip(const guchar *gzip_data, size_t *len_ptr)
+{
+	gsize gzip_data_len	= *len_ptr;
+	z_stream zstr;
+	int gzip_err = 0;
+	gchar *data_buffer;
+	gulong gzip_len = G_MAXUINT16;
+	GString *output_string = NULL;
+
+	data_buffer = g_new0(gchar, gzip_len);
+
+	zstr.next_in = NULL;
+	zstr.avail_in = 0;
+	zstr.zalloc = Z_NULL;
+	zstr.zfree = Z_NULL;
+	zstr.opaque = 0;
+	gzip_err = inflateInit2(&zstr, MAX_WBITS+32);
+	if (gzip_err != Z_OK)
+	{
+		g_free(data_buffer);
+		purple_debug_error("matrixprpl", "no built-in gzip support in zlib\n");
+		return NULL;
+	}
+
+	zstr.next_in = (Bytef *)gzip_data;
+	zstr.avail_in = gzip_data_len;
+
+	zstr.next_out = (Bytef *)data_buffer;
+	zstr.avail_out = gzip_len;
+
+	gzip_err = inflate(&zstr, Z_SYNC_FLUSH);
+
+	if (gzip_err == Z_DATA_ERROR)
+	{
+		inflateEnd(&zstr);
+		inflateInit2(&zstr, -MAX_WBITS);
+		if (gzip_err != Z_OK)
```

Comment:
I don't understand this. We know that `gzip_err == Z_DATA_ERROR` from four lines above. Why are we now comparing it with `Z_OK`?

Reply:
I'd like to see @EionRobb  comment on that. But clearly the code is bad. I'll drop this part.

Reply:
I guess correct line would be:

```
gzip_err = inflateInit2(&zstr, -MAX_WBITS);
```

Reply:
done